packages/backend/src/modules/tvl2/utils/SyncOptimizer.ts

```diff
@@ -14,10 +15,17 @@ export class SyncOptimizer {
   ) {}
 
   shouldTimestampBeSynced(timestamp: UnixTime) {
-    return timestamp.equals(this.getTimestampToSync(timestamp))
+    return timestamp.equals(this.getOptimizedTimestamp(timestamp))
   }
 
-  getTimestampToSync(timestamp: UnixTime): UnixTime {
+  getTimestampToSync(from: number, to: number): UnixTime {
+    const timestamp = this.getOptimizedTimestamp(new UnixTime(from))
+    assert(timestamp.lte(new UnixTime(to)), 'Invalid range')
```

Comment:
I'm not sure if that's good approach. `getTimestampToSync` is used in the `update` function and if it throws, `update` will be called with the same arguments again. I understand we're introducing another assumption that there's always a timestamp to sync between `to` and `from`, but this depends on how parent indexers are functioning. I'm not saying it's bad, just something to keep in mind.